### PR TITLE
Fixed bug in GuiceViewProvider getViewName() and added test for bug

### PR DESCRIPTION
**Pull request by peergynt**
As a side note, new Reflections("com.vaadin.guice.server.testClasses") in ScopeTestBase does not seem right.
Looks like the package is now "com.vaadin.guice.testClasses".

### CODE REVIEW

File: src/main/java/com/vaadin/guice/server/GuiceViewProvider.java [hunk 1]
```diff
@@ -16,6 +16,7 @@
 package com.vaadin.guice.server;
 
 import com.google.common.collect.ImmutableMap;
+import com.google.gwt.thirdparty.guava.common.collect.ImmutableSortedMap;
```
Comment (berndhopp) [inline src/main/java/com/vaadin/guice/server/GuiceViewProvider.java]: should be com.google.common.collect.ImmutableSortedMap I guess

File: src/main/java/com/vaadin/guice/server/GuiceViewProvider.java [hunk 2]
```diff
@@ -50,19 +52,20 @@
...
     private final Map<String, Class<? extends View>> viewNamesToViewClassesMap;
     private final Map<VaadinSession, Map<String, View>> viewsBySessionMap;
-    private final Set<String> viewNames;
+    private final NavigableSet<String> viewNames;
 
     public GuiceViewProvider(Set<Class<? extends View>> viewClasses) {
 
         viewNamesToViewClassesMap = scanForViews(viewClasses);
-        viewNames = viewNamesToViewClassesMap.keySet();
+        viewNames = ImmutableSortedMap.copyOf(viewNamesToViewClassesMap).keySet();
```
Comment (berndhopp) [inline src/main/java/com/vaadin/guice/server/GuiceViewProvider.java:66]: why create a copy? viewNamesToViewClassesMap is ImmutableMap anyway.

File: src/main/java/com/vaadin/guice/server/GuiceViewProvider.java [hunk 3]
```diff
@@ -103,17 +106,18 @@ public GuiceViewProvider(Set<Class<? extends View>> viewClasses) {
...
-
-            if (viewAndParameters.startsWith(viewName)) {
-                return viewName;
-            }
+        if (viewAndParameters == null) {
+            return null;
         }
-
-        return viewAndParameters;
+        String viewName = viewNames.floor(viewAndParameters);
```
Comment (berndhopp) [inline src/main/java/com/vaadin/guice/server/GuiceViewProvider.java:118]: a comment to clarify why and how this works would be good

File: src/main/java/com/vaadin/guice/server/GuiceViewProvider.java (hunk 1 above)
Comment (peergynt) [inline src/main/java/com/vaadin/guice/server/GuiceViewProvider.java]: Agreed, fixed

File: src/main/java/com/vaadin/guice/server/GuiceViewProvider.java (hunk 2 above)
Comment (peergynt) [inline src/main/java/com/vaadin/guice/server/GuiceViewProvider.java:66]: Added some comments in the code. The purpose of the line is merely to get a sorted/navigable set of view names.

File: src/main/java/com/vaadin/guice/server/GuiceViewProvider.java (hunk 3 above)
Comment (peergynt) [inline src/main/java/com/vaadin/guice/server/GuiceViewProvider.java:118]: Added comment.

**Comment by berndhopp** [comment]
good work

**Comment by berndhopp** [comment]
@peergynt can we have a hangout to discuss further development?

regards

**Pull request merged**